From 159b69da38e11bf5e8e22a04462f80dfd5fb39cf Mon Sep 17 00:00:00 2001
From: "liang.he" <liang.he@intel.com>
Date: Wed, 2 Apr 2025 07:13:53 +0800
Subject: [PATCH] Update cmake min to 3.14 (#4175)
URL: https://github.com/bytecodealliance/wasm-micro-runtime/commit/159b69da38e11bf5e8e22a04462f80dfd5fb39cf.patch

3.14 is used and tested by linux mini-product

to fix

```
CMake Error at CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

Edited to apply to fluent-bit source tree by Jaco Kroon <jaco@uls.co.za>

---
 CMakeLists.txt                                                  | 2 +-
 build-scripts/esp-idf/README.md                                 | 2 +-
 core/iwasm/libraries/lib-socket/lib_socket_wasi.cmake           | 2 +-
 doc/build_wasm_app.md                                           | 2 +-
 product-mini/app-samples/hello-world-cmake/CMakeLists.txt       | 2 +-
 product-mini/platforms/esp-idf/CMakeLists.txt                   | 2 +-
 product-mini/platforms/freebsd/CMakeLists.txt                   | 2 +-
 product-mini/platforms/linux-sgx/CMakeLists.txt                 | 2 +-
 product-mini/platforms/linux-sgx/CMakeLists_minimal.txt         | 2 +-
 product-mini/platforms/riot/CMakeLists.txt                      | 2 +-
 product-mini/platforms/vxworks/CMakeLists.txt                   | 2 +-
 product-mini/platforms/windows/CMakeLists.txt                   | 2 +-
 product-mini/platforms/zephyr/simple/CMakeLists.txt             | 2 +-
* product-mini/platforms/zephyr/user-mode/CMakeLists.txt          | 2 +-
* .../platforms/zephyr/user-mode/lib-wamr-zephyr/CMakeLists.txt   | 2 +-
 samples/bh-atomic/CMakeLists.txt                                | 2 +-
 samples/file/CMakeLists.txt                                     | 2 +-
 samples/file/wasm-app/CMakeLists.txt                            | 2 +-
 samples/mem-allocator/CMakeLists.txt                            | 2 +-
 samples/multi-module/wasm-apps/CMakeLists.txt                   | 2 +-
 samples/native-lib/wasm-app/CMakeLists.txt                      | 2 +-
 samples/sgx-ra/CMakeLists.txt                                   | 2 +-
 samples/sgx-ra/wasm-app/CMakeLists.txt                          | 2 +-
 samples/socket-api/wasm-src/CMakeLists.txt                      | 2 +-
 test-tools/binarydump-tool/CMakeLists.txt                       | 2 +-
 .../wamr-ide/VSCode-Extension/resource/scripts/CMakeLists.txt   | 2 +-
 26 files changed, 26 insertions(+), 26 deletions(-)

diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/CMakeLists.txt b/CMakeLists.txt
index e6d1c840a7..551991f891 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 3.0)
+cmake_minimum_required (VERSION 3.14)
 
 option(BUILD_SHARED_LIBS "Build using shared libraries" OFF)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/build-scripts/esp-idf/README.md b/build-scripts/esp-idf/README.md
index 6bec45d1e7..25fe491ea6 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/build-scripts/esp-idf/README.md
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/build-scripts/esp-idf/README.md
@@ -11,7 +11,7 @@ You can build an ESP-IDF project with wasm-micro-runtime as a component:
 - In the newly created project folder edit the `CMakeList.txt`:
 
   ```
-  cmake_minimum_required(VERSION 3.5)
+  cmake_minimum_required(VERSION 3.14)
 
   include($ENV{IDF_PATH}/tools/cmake/project.cmake)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/core/iwasm/libraries/lib-socket/lib_socket_wasi.cmake b/core/iwasm/libraries/lib-socket/lib_socket_wasi.cmake
index 8ddddffeba..c6a4430f7f 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/core/iwasm/libraries/lib-socket/lib_socket_wasi.cmake
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/core/iwasm/libraries/lib-socket/lib_socket_wasi.cmake
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.8...3.16)
+cmake_minimum_required (VERSION 3.14)
 
 project(socket_wasi_ext LANGUAGES C)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/doc/build_wasm_app.md b/doc/build_wasm_app.md
index ef9bea9e24..2cea780690 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/doc/build_wasm_app.md
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/doc/build_wasm_app.md
@@ -275,7 +275,7 @@ You can cross compile your project by using the toolchain provided by WAMR.
 Assume the original `CMakeLists.txt` for `test.c` likes below:
 
 ``` cmake
-cmake_minimum_required (VERSION 3.5)
+cmake_minimum_required (VERSION 3.14)
 project(hello_world)
 add_executable(hello_world test.c)
 ```
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/app-samples/hello-world-cmake/CMakeLists.txt b/product-mini/app-samples/hello-world-cmake/CMakeLists.txt
index b41fe0a515..6131b69772 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/app-samples/hello-world-cmake/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/app-samples/hello-world-cmake/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 3.5)
+cmake_minimum_required (VERSION 3.14)
 project(hello_world)
 
 set (CMAKE_C_FLAGS          "${CMAKE_C_FLAGS} -O3 -Wno-unused-command-line-argument")
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/esp-idf/CMakeLists.txt b/product-mini/platforms/esp-idf/CMakeLists.txt
index 8472df8dd8..309949a20b 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/esp-idf/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/esp-idf/CMakeLists.txt
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 # from ESP-IDF 4.0 examples/build_system/cmake/idf_as_lib
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.14)
 
 include($ENV{IDF_PATH}/tools/cmake/project.cmake)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/freebsd/CMakeLists.txt b/product-mini/platforms/freebsd/CMakeLists.txt
index 5640a384a0..fccc523236 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/freebsd/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/freebsd/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
 
 project (iwasm)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/linux-sgx/CMakeLists.txt b/product-mini/platforms/linux-sgx/CMakeLists.txt
index 927e6f4592..bc3bde5657 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/linux-sgx/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/linux-sgx/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
 
 project (iwasm)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/linux-sgx/CMakeLists_minimal.txt b/product-mini/platforms/linux-sgx/CMakeLists_minimal.txt
index a29dbd69c1..5104769d73 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/linux-sgx/CMakeLists_minimal.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/linux-sgx/CMakeLists_minimal.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
 
 project (iwasm)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/riot/CMakeLists.txt b/product-mini/platforms/riot/CMakeLists.txt
index c32a0b9777..cc98146c57 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/riot/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/riot/CMakeLists.txt
@@ -2,7 +2,7 @@
 # Copyright (C) 2020 TU Bergakademie Freiberg Karl Fessel
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.8.2)
+cmake_minimum_required(VERSION 3.14)
 
 set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/vxworks/CMakeLists.txt b/product-mini/platforms/vxworks/CMakeLists.txt
index dd03cb356f..3f08a72ad0 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/vxworks/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/vxworks/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
 
 project (iwasm)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/windows/CMakeLists.txt b/product-mini/platforms/windows/CMakeLists.txt
index 5fcc276a19..9ec5d34152 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/windows/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/windows/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
 
 project (iwasm C ASM CXX)
 # set (CMAKE_VERBOSE_MAKEFILE 1)
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/zephyr/simple/CMakeLists.txt b/product-mini/platforms/zephyr/simple/CMakeLists.txt
index 78dd322858..46dc3ea0c5 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/zephyr/simple/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/product-mini/platforms/zephyr/simple/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.8.2)
+cmake_minimum_required(VERSION 3.14)
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(wamr)
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/bh-atomic/CMakeLists.txt b/samples/bh-atomic/CMakeLists.txt
index f690527421..177cf2c5eb 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/bh-atomic/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/bh-atomic/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2023 Midokura Japan KK.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.14)
 project(bh_atomic)
 
 string (TOLOWER ${CMAKE_HOST_SYSTEM_NAME} WAMR_BUILD_PLATFORM)
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/file/CMakeLists.txt b/samples/file/CMakeLists.txt
index c3a69ccc86..fca2003765 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/file/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/file/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2022 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.14)
 project(file)
 
 ################ wasm application ###############
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/file/wasm-app/CMakeLists.txt b/samples/file/wasm-app/CMakeLists.txt
index 4af87a3fdd..f63485ca03 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/file/wasm-app/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/file/wasm-app/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2022 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.14)
 project(wasm-app)
 
 set (WAMR_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../..)
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/mem-allocator/CMakeLists.txt b/samples/mem-allocator/CMakeLists.txt
index f157dfbde5..9f3e49db9c 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/mem-allocator/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/mem-allocator/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2023 Midokura Japan KK.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.14)
 project(mem_allocator_create)
 
 string (TOLOWER ${CMAKE_HOST_SYSTEM_NAME} WAMR_BUILD_PLATFORM)
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/multi-module/wasm-apps/CMakeLists.txt b/samples/multi-module/wasm-apps/CMakeLists.txt
index 8dbb52686c..6460a99e67 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/multi-module/wasm-apps/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/multi-module/wasm-apps/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.8...3.16)
+cmake_minimum_required (VERSION 3.14)
 project(wasm-apps)
 
 message(CHECK_START "Detecting WABT")
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/native-lib/wasm-app/CMakeLists.txt b/samples/native-lib/wasm-app/CMakeLists.txt
index ffcd9005ab..e5bea6010b 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/native-lib/wasm-app/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/native-lib/wasm-app/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.14)
 project(wasm-app)
 
 set (WAMR_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../..)
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/sgx-ra/CMakeLists.txt b/samples/sgx-ra/CMakeLists.txt
index 3f5f23e97c..48ff9ee569 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/sgx-ra/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/sgx-ra/CMakeLists.txt
@@ -2,7 +2,7 @@
 # Copyright (c) 2020-2021 Alibaba Cloud
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.1.4)
+cmake_minimum_required(VERSION 3.14)
 project(sgx-ra)
 
 ################ runtime settings  ##############
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/sgx-ra/wasm-app/CMakeLists.txt b/samples/sgx-ra/wasm-app/CMakeLists.txt
index afba7dfb6d..93907b7f0d 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/sgx-ra/wasm-app/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/sgx-ra/wasm-app/CMakeLists.txt
@@ -2,7 +2,7 @@
 # Copyright (c) 2020-2021 Alibaba Cloud
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.14)
 project(wasm-app)
 
 set (WAMR_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../..)
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/socket-api/wasm-src/CMakeLists.txt b/samples/socket-api/wasm-src/CMakeLists.txt
index 8f11e0a6b4..38e8139168 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/samples/socket-api/wasm-src/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/samples/socket-api/wasm-src/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 2.8...3.18)
+cmake_minimum_required(VERSION 3.14)
 project(socket_api_sample_wasm_app)
 
 message(CHECK_START "Detecting WABT")
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/test-tools/binarydump-tool/CMakeLists.txt b/test-tools/binarydump-tool/CMakeLists.txt
index d322b42b66..744f72c3c8 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/test-tools/binarydump-tool/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/test-tools/binarydump-tool/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
 
 project(binarydump)
 
diff --git a/lib/wasm-micro-runtime-WAMR-1.3.3/test-tools/wamr-ide/VSCode-Extension/resource/scripts/CMakeLists.txt b/test-tools/wamr-ide/VSCode-Extension/resource/scripts/CMakeLists.txt
index 81d998bc83..a6298c5e10 100644
--- a/lib/wasm-micro-runtime-WAMR-1.3.3/test-tools/wamr-ide/VSCode-Extension/resource/scripts/CMakeLists.txt
+++ b/lib/wasm-micro-runtime-WAMR-1.3.3/test-tools/wamr-ide/VSCode-Extension/resource/scripts/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
 
 project(Main)
 
